### PR TITLE
fix: modified_content on row documents

### PR DESCRIPTION
**Pull request by sbor23**

### CODE REVIEW

File: caluma/caluma_form/models.py [hunk 1]
```diff
@@ -295,11 +295,13 @@ def copy(self, family=None, user=None):
...
 
+    def is_root(self):
+        return self == self.family
+
     @cached_property
     def last_modified_answer(self):
-        return (
-            Answer.objects.filter(document__family=self).order_by("modified_at").last()
-        )
+        _filter = "document__family" if self.is_root() else "document"
```
Comment (winged) [inline caluma/caluma_form/models.py:303]: It took me quite a while to understand why this is required. I think a comment and an extended commit message would be very useful.

Somethings along the lines of "for row documents, we want the most recently changed answer within this document, not traversing to the parent/root document. Only for root documents should we look at all answers, including rows"

**Review by winged — CHANGES_REQUESTED** [review]
Nice catch! But please explain in code & commit message

File: caluma/caluma_form/models.py [hunk 2]
```diff
@@ -295,11 +295,13 @@ def copy(self, family=None, user=None):
...
+        return Answer.objects.filter(**{_filter: self}).order_by("-modified_at").first()
```
Comment (anehx) [inline caluma/caluma_form/models.py:304]: Is it really necessary to differentiate? Can't we just do `Answer.objects.filter(Q(document=self) | Q(document__family=self)).order_by("modified_at").last()`?

File: caluma/caluma_form/models.py (hunk 1 above)
Comment (sbor23) [reply caluma/caluma_form/models.py]: You are right, added comment and commit message.b

File: caluma/caluma_form/models.py (hunk 2 above)
Comment (sbor23) [reply caluma/caluma_form/models.py]: That should also work and is much nicer to read! I think we could save a join by differentiating but it's not worth it.

**Review by anehx — APPROVED** [review]

**Review by winged — APPROVED** [review]

**Comment by anehx** [comment]
@sbor23 can we merge and release this?

**Pull request merged**